File: Algorithm_Study/Programmers/level2/피로도.py
import itertools

def solution(k, dungeons):
    answer = -1
    combi=[]
    for i in range(len(dungeons)):
        combi.append(i)

    combi=list(itertools.permutations(combi, len(dungeons)))
    for i in combi:
        my_hp=k
        cnt=0
        for j in i:
            if dungeons[j][0]<=my_hp:
                my_hp-=dungeons[j][1]
                cnt+=1

        answer=max(cnt,answer)




    return answer

File: Algorithm_Study/Programmers/level2/피로도.js
let answer=-1
const dfs=(current_p,dungeons,depth,visited)=>{
    for(let i=0;i<dungeons.length;i++){
        const [min_p,p]=dungeons[i]
        if(visited[i])continue
        if(current_p>=min_p){
            visited[i]=true
            dfs(current_p-p,dungeons,depth+1,visited)
            visited[i]=false
        }
    }
    answer=Math.max(answer,depth)
    return
}
function solution(k, dungeons) {
    let visited=new Array(dungeons.length).fill(false)
    dfs(k,dungeons,0,visited)
    return answer;
}
solution(80,[[80,20],[50,40],[30,10]])